specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,262 @@
+- declarations of functions used for [output recording](#output-recording)
+- [attributes](#attributes)
+
+The human readable LLVM IR for the bitcode can be obtained using standard [LLVM
+tools](https://llvm.org/docs/CommandGuide/llvm-dis.html). For the purpose of
+clarity, this specification contains examples of the human readable IR emitted
+by [LLVM 13](https://releases.llvm.org/13.0.1/docs/LangRef.html). While the
+bitcode representation is portable and usually backward compatible, there may be
+visual differences in the human readable format depending on the LLVM version.
+This differences are irrelevant when using standard tools to load, manipulate,
+and/or execute bitcode.
+
+The code below illustrates how a simple program looks like within a Base Profile
+representation:
+
+```llvm
+; type definitions
+
+%Result = type opaque
+%Qubit = type opaque
+
+; global constants (labels for output recording)
+
+@0 = internal constant [16 x i8] c"label_schema_id\00"
+@1 = internal constant [3 x i8] c"r1\00"
+@2 = internal constant [3 x i8] c"r2\00"
+
+; entry point definition
+
+define i64 @Entry_Point_Name() #0 {
+entry:
+
+  ; calls to QIS functions
+  tail call void @__quantum__qis__h__body(%Qubit* null)
+  tail call void @__quantum__qis__cnot__body(%Qubit* null, %Qubit* inttoptr (i64 1 to %Qubit*))
+  tail call void @__quantum__qis__mz__body(%Qubit* null, %Result* null)
+  tail call void @__quantum__qis__mz__body(%Qubit* inttoptr (i64 1 to %Qubit*), %Result* inttoptr (i64 1 to %Result*))
+
+  ; calls to record the program output
+  tail call void @__quantum__rt__initialize_record_output(i8* getelementptr inbounds ([16 x i8], [16 x i8]* @0, i32 0, i32 0))
+  tail call void @__quantum__rt__tuple_record_output(i64 2, i8* null)
+  tail call void @__quantum__rt__result_record_output(%Result* null, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @1, i32 0, i32 0))
+  tail call void @__quantum__rt__result_record_output(%Result* inttoptr (i64 1 to %Result*), i8* getelementptr inbounds ([3 x i8], [3 x i8]* @2, i32 0, i32 0))
+
+  ret i64 0
+}
+
+; declarations of QIS functions
+
+declare void @__quantum__qis__h__body(%Qubit*)
+
+declare void @__quantum__qis__cnot__body(%Qubit*, %Qubit*)
+
+declare void @__quantum__qis__mz__body(%Qubit*, %Result*)
+
+; declarations of functions used for output recording
+
+declare void @__quantum__rt__initialize_record_output(i8*)
+
+declare void @__quantum__rt__tuple_record_output(i64, i8*)
+
+declare void @__quantum__rt__result_record_output(%Result*, i8*)
+
+; attributes
+
+attributes #0 = { "entry_point" "required_qubits"="2" "required_results"="2" }
+```
+
+TODO: do we need additional restrictions for output recording? Like e.g. it
+needs to be a dedicated block, or output recording functions can only occur in
+the last block of the entry point?
+
+TODO: do we need a `__quantum__rt__initialize` and `__quantum__rt__finalize`?
+TODO: do measurements need to be at the end of the quantum computations, or
```

Comment:
I think we concluded that measurement indeed need to be at the end also within the IR (rather than going with just the requirement that qubits are not used after measurements). I heavily edited this - please take another look; I believe your comment should be addressed.